README.md

```diff
@@ -1,2 +1,240 @@
+            case 1: this.UserID = (string)fieldValue; break;
+            case 2: this.RegionID = (string)fieldValue; break;
+            case 3: this.Gender = (string)fieldValue; break;
+            default: throw new AvroRuntimeException("Bad index " + fieldPos + " in Put()");
+        };
+    }
+}
+
+
+public static void User(
+    [KafkaTrigger("BrokerList", "users", ValueType=typeof(UserRecord), ConsumerGroup = "myGroupId")] KafkaEventData[] kafkaEvents,
+    ILogger logger)
+{
+    foreach (var kafkaEvent in kafkaEvents)
+    {
+        var user = (UserRecord)kafkaEvent.Value;
+        logger.LogInformation($"{JsonConvert.SerializeObject(kafkaEvent.Value)}");
+    }
+}
+```
+
+*Using Avro Generic*
+
+1. In `KafkaTrigger` attribute set the value of `AvroSchema` to the string representation of it.
+1. The parameter type used with the trigger must be of type `KafkaEventData`. The value of `KafkaEventData.Value` will be of the type `GenericRecord`.
+
+The sample function contains 1 consumer using avro generic. Check the class `AvroGenericTriggers`
+
+```csharp
+public static class AvroGenericTriggers
+{
+      const string PageViewsSchema = @"{
+  ""type"": ""record"",
+  ""name"": ""pageviews"",
+  ""namespace"": ""ksql"",
+  ""fields"": [
+    {
+      ""name"": ""viewtime"",
+      ""type"": ""long""
+    },
+    {
+      ""name"": ""userid"",
+      ""type"": ""string""
+    },
+    {
+      ""name"": ""pageid"",
+      ""type"": ""string""
+    }
+  ]
+}";
+
+[FunctionName(nameof(PageViews))]
+public static void PageViews(
+    [KafkaTrigger("BrokerList", "pageviews", AvroSchema = PageViewsSchema, ConsumerGroup = "myGroupId")] KafkaEventData kafkaEvent,
+    ILogger logger)
+{
+    if (kafkaEvent.Value is GenericRecord genericRecord)
+    {
+        // Get the field values manually from genericRecord
+    }
+}
+```
+
+#### Protobuf Binding Support
+
+Protobuf is supported in the trigger based on the `Google.Protobuf` nuget package. To consume a topic that is using protobuf as serialization set the ValueType to be of a type that implements `Google.Protobuf.IMessage`. The sample producer has a producer for topic `protoUser` (must be created). The sample function has a trigger handler for this topic in class `ProtobufTriggers`.
+
+```csharp
+public static class ProtobufTriggers
+{
+    [FunctionName(nameof(ProtobufUser))]
+    public static void ProtobufUser(
+        [KafkaTrigger("BrokerList", "protoUser", ValueType=typeof(ProtoUser), ConsumerGroup = "myGroupId")] KafkaEventData[] kafkaEvents,
+        ILogger logger)
+    {
+        foreach (var kafkaEvent in kafkaEvents)
+        {
+            var user = (ProtoUser)kafkaEvent.Value;
+            logger.LogInformation($"{JsonConvert.SerializeObject(user)}");
+        }
+    }
+}
+```
+
+### Output Binding
+
+[TODO: add output binding documentation]
+
+## Quickstart
+
+For samples take a look at the [samples folder](./samples).
+
+## License

```

Comment:
there already is a specific License file in the root that we should be referring people to. This was added for us when I generated the repo using our opensource tooling.

Reply:
Should we just link to it? As:

## License
[MIT](LICENSE)

Or should we just not mention in index page?

Reply:
I'd just leave it out of the index page personally. But don't mind linking back to the License file either. As long as we don't create a different license file.

Reply:
I will remove.